### PR TITLE
Cleanup

### LINKED ISSUE #166: Drop un-used CSS
Much of the CSS currently in content-kit can now be dropped. Only the bits pertaining to highlights and the blank-line widget should remain.

### PR DESCRIPTION
**Pull request by bantic**
* removes all the files in `src/js/commands/` — any previous usage of these is replaced with usage of the `postEditor` instead
  * removes the "edit" state on image sections. This is not used by anyone (that I know of). The image section itself may be a target for removal later.
  * removes all `view`s except the base view and tooltip view
  * removes a bunch of then- and now-unused functions in the `utils`
  * remove several unused test helper methods and files
  * remove unused CSS (fixes #166)
  * remove `selection*` events from the editor (the public API is to register callbacks for `cursorDidChange`, instead. The internal views that used to subscribe to these events are gone now).

Tested with ember-content-kit and by hand with the demo app

cc @mixonic

### CODE REVIEW

File: src/js/cards/image.js [hunk 1]
```diff
@@ -1,96 +1,17 @@
...
-        window.alert('There was a problem uploading the image: '+error);
-        failure();
-      }
-    }
-  });
-}
+const placeholderSrc = "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAOEAAADhCAMAAAAJbSJIAAAAMFBMVEXp7vG6vsHm6+63u77Hy868wMPe4+bO09bh5unr8fTR1djAxMfM0NPX3N/c4eTBxcjXRf5TAAACh0lEQVR4nO3b6ZKqMBSFUSQMYZL3f9tbBq/NEEDiqUqOfusn1ZXKbjcQlGQZAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACC6RkbsGHuabChEtHmiGYfS3EQYM+Sxw/gMQvmcNnYaj6oTDHi73WPn2eqnj9B8zo3TJXcq5uNjXmVff86VwSR3JtryMa1BYqi7S1hJDCVpSigyLcGhJJEwzlCSNtPKrbVhVwsdCfOhH7uuaG3ARV9DwsaOzxt3N1yPqCHhvXytTUz92VDpmE/LLhZwl++R6Sds6sUa/PL6K/2E2fIhw1xdRKefsFolrPc+xNx/N0k/4fpBsdhL2HfeiN+TsDCms8dDpeRyS3P3QDl6Iqaf8L0rTf+80m6Lmn7Ct+4Wxf+/2RY1/YRv3PHz/u+fsCmqgoTnq7Z+8SGviqoh4dnKu1ieqauiakh4/PQ0r6ivqDoSHj0B97eNRVG1JNxV+L4bnxdVecJtRTdFVZ7QU9F1UXUn9FZ0VVRlCav5ob2KLouqKmFjy676u2HsVnRRVFUJq3J+8KCi86IqSthMvyl209Hjijqm3RsqAZ5pNfa5PJ2KelJRjQmr1/r7cfy0ouoSNvOfvbvhvKLaEr4qOin9kTQnrN7LpDZhE/Zmhp6Eq4p+YcKgiipKGFhRRQkDK6ooYfgLbiSMioQkJGF8P5XwHv4O+7AaKiXzaeXh1kMl5AffTUxiKEm/krD94BR8Gdxl1fceSlR58ZhXKbEpyD2amNiBtmrJLTMHL1LF8/rpXkSZXEmz8K8uvAFFNm6Iq0aBLUFOmeCuJ6exrcCmoLpN7kYx891bSAEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADgh/wDdr8peyRHLogAAAAASUVORK5CYII=";
```
Comment (mixonic) [inline src/js/cards/image.js]: Keeping this in its own file would have been nice

Comment (bantic) [inline src/js/cards/image.js]: what about having it in `utils/`? Having a file called `cards/placeholder-image.js` felt odd.

File: src/js/utils/dom-utils.js [hunk 2]
```diff
@@ -53,24 +35,6 @@ function clearChildNodes(element) {
   }
 }
 
-// walks DOWN the dom from node to childNodes, returning the element
-// for which `conditionFn(element)` is true
-function walkDOMUntil(topNode, conditionFn=() => {}) {
```
Comment (bantic) [inline src/js/utils/dom-utils.js]: this function is only used by the test helpers, so it was moved there.

File: src/js/editor/editor.js [hunk 3]
```diff
@@ -308,7 +301,7 @@ class Editor {
    */
   handleInput() {
     this.reparse();
-    this.trigger('update');
+    this.didUpdate();
```
Comment (mixonic) [inline src/js/editor/editor.js]: :fireworks: yay

File: src/js/index.js [hunk 4]
```diff
@@ -1,11 +1,9 @@
 import Editor from './editor/editor';
-import BoldCommand from './commands/bold';
 import ImageCard from './cards/image';
 
 const ContentKit = {
   Editor,
-  ImageCard,
-  BoldCommand
```
Comment (mixonic) [inline src/js/index.js]: This was used in old Typeset, but can go. :+1:

File: src/js/cards/image.js (hunk 1 above)
Comment (mixonic) [inline src/js/cards/image.js]: `utils/` seems great :+1:

File: src/js/utils/dom-utils.js [hunk 5]
```diff
@@ -156,16 +100,12 @@ function parseHTML(html) {
 }
 
 export {
-  detectParentNode,
```
Comment (mixonic) [inline src/js/utils/dom-utils.js]: surprised this is not used, but it is not!

**Comment by mixonic** [comment]
:clap: Thank you sir!

**Comment by bantic** [comment]
moved the placeholder img src to `utils/` and squashed these commits. will merge when the tests are green

**Pull request merged**